Data/HashMap/Base.hs

```diff
@@ -199,11 +199,19 @@ instance Foldable.Foldable (HashMap k) where
     foldr f = foldrWithKey (const f)
 
 #if __GLASGOW_HASKELL__ >= 711
+-- | '<>' = 'union'
+--
+-- If a key occurs in both maps, the mapping from the first will be the mapping in the result.
```

Comment:
Could you add an example in a collapsible section? You can do that with:

```haskell
-- ==== __Examples__
--
-- example here
--
```

Reply:
Shouldn't there be examples in the `union` documentation instead?

Reply:
Either that with words pointing them to the union for examples, or have the examples in both places.

Reply:
I guess I can, though I'm confused why that is a blocker to this PR?

Reply:
I'm going to disagree with both of you. Long Haddock documentation for instances is (in my experience) always formatted horribly. Therefore, I think the right approach is to limit the documentation string to

```haskell
 +-- | '<>' = 'union' 
```

and let users follow the link to find out what that means.